src/undo/durable_move.py

```
@@ -0,0 +1,388 @@
+        FileNotFoundError: If *src* doesn't exist.
+        OSError: Any other OS-level failure (permission denied, disk
+            full, etc.). The journal will contain a ``started`` or
+            ``copied`` entry for the interrupted operation;
+            :func:`sweep` on next startup cleans up.
+    """
+    src = Path(src)
+    dst = Path(dst)
+    journal = Path(journal)
+
+    if not src.exists() and not src.is_symlink():
+        raise FileNotFoundError(f"Source does not exist: {src}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)